internal/controller/ngrok/ngroktrafficpolicy_controller.go

```diff
@@ -63,7 +66,25 @@ type NgrokTrafficPolicyReconciler struct {
 func (r *NgrokTrafficPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = log.FromContext(ctx)
 
-	err := r.Driver.SyncEdges(ctx, r.Client)
+	policy := &ngrokv1alpha1.NgrokTrafficPolicy{}
+	if err := r.Get(ctx, req.NamespacedName, policy); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	parsedTrafficPolicy, err := util.NewTrafficPolicyFromJson(policy.Spec.Policy)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if parsedTrafficPolicy.IsLegacyPolicy() {
+		r.Recorder.Eventf(policy, v1.EventTypeWarning, events.PolicyDeprecation, "Traffic Policy is using legacy directions: ['inbound', 'outbound']. Update to new phases: ['on_tcp_connect', 'on_http_request', 'on_http_response']")
+	}
+
+	if parsedTrafficPolicy.Enabled() != nil {
+		r.Recorder.Eventf(policy, v1.EventTypeWarning, events.PolicyDeprecation, "Traffic Policy has 'enabled' set. This is a legacy option that will stop being supported soon.")
+	}
+
+	err = r.Driver.SyncEdges(ctx, r.Client)
```

Comment:
we may want to consider moving the recorder stuff to after the driver syncs. at the very least, not error out if the parsing, etc stuff doesn't work for the purposes of events. If this event stuff fails, recorder functions will also probably certainly fail. however, we may want to not take a risk there.